src/evaluate/evaluate.ts

```diff
@@ -45,8 +45,8 @@ export class Evaluator {
   constructor(public readonly context: EvaluationContext) {}
 
   public evaluateTemplate() {
-    this.evaluateMetadata();
     this.evaluateParameters();
+    this.evaluateMetadata();
```

Comment:
Is the change of evaluation order a functional one or purely aesthetic?

If it's functional, I'd like us to add an explicit unit test for it.

Reply:
Yes, it's functional. Metadata may reference parameters, so the parameters have to be evaluated first. I'll add a test.